build: acceptance tests crash with "/usr/bin/ld: .eh_frame_hdr table[1703] FDE at 00000000022c9538 overlaps table[1704] FDE at 00000000022d0548."

I've tried to run the acceptance tests with "make acceptance" under Centos7.2, under Debian8.5, and under Ubuntu14.04. In all three cases, the "make acceptance" attempt crashes with a complaint about c++ exception handling frames. I strongly suspect this has something to do with the https://hub.docker.com/r/cockroachdb/builder/  image itself, because "make install" under Debian8.5 would do the exact same thing, with no Docker image involved. (Note that the docker image is also using Debian8.5.)

full transcript below, as one example, from the Ubuntu14.04 case:
I'm using go1.6.2.
~~~
jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ ls
acceptance  build	cli	cloud  config		docs	  GLOCKFILE  keys  LICENSE  Makefile   resource  rpc	   server  storage   testutils	ui
base	    circle.yml	client	cmd    CONTRIBUTING.md	githooks  gossip     kv    main.go  README.md  roachpb	 security  sql	   STYLE.md  ts		util
jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ make acceptance
Unable to find image 'cockroachdb/builder:20160611-170214' locally
20160611-170214: Pulling from cockroachdb/builder
[0B
[0B
[0B
[0B
[0B
[0B
[0B
[0B
[1BDigest: sha256:c656d53a35e3ff68e7c14e5fa6e8314575e7133528458fa2ca2526ab7a8d28ea
Status: Downloaded newer image for cockroachdb/builder:20160611-170214
GOPATH set to /go
/go/bin added to PATH
go install -v -tags clockoffset
github.com/cockroachdb/cockroach
# github.com/cockroachdb/cockroach
/usr/local/go/pkg/tool/linux_amd64/link: running g++ failed: exit status 1
/usr/bin/ld: .eh_frame_hdr table[1703] FDE at 00000000022c9538 overlaps table[1704] FDE at 00000000022d0548.
/usr/bin/ld: .eh_frame_hdr table[1704] FDE at 00000000022d0548 overlaps table[1705] FDE at 00000000022d10c8.
/usr/bin/ld: .eh_frame_hdr table[1705] FDE at 00000000022d10c8 overlaps table[1706] FDE at 00000000022d2f08.
/usr/bin/ld: .eh_frame_hdr table[1706] FDE at 00000000022d2f08 overlaps table[1707] FDE at 00000000022d3dd8.
/usr/bin/ld: .eh_frame_hdr table[1707] FDE at 00000000022d3dd8 overlaps table[1708] FDE at 00000000022d42e0.
...
 <tons more of the same, omitted to avoid overwhelming github's editor>
...
/usr/bin/ld: .eh_frame_hdr table[11990] FDE at 00000000023436f8 overlaps table[11991] FDE at 0000000002344fa8.
/usr/bin/ld: .eh_frame_hdr table[11993] FDE at 0000000002343760 overlaps table[11994] FDE at 0000000002345058.
/usr/bin/ld: .eh_frame_hdr table[11998] FDE at 0000000002343910 overlaps table[11999] FDE at 0000000002345088.
/usr/bin/ld: .eh_frame_hdr table[12001] FDE at 0000000002343b38 overlaps table[12002] FDE at 0000000002344b28.
collect2: error: ld returned 1 exit status

Makefile:88: recipe for target 'install' failed
make: *** [install] Error 2
make: *** [acceptance] Error 2
jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ uname -a 
Linux crdev 3.13.0-85-generic #129-Ubuntu SMP Thu Mar 17 20:50:15 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.4 LTS"
jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ docker version
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64
jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ git log
WARNING: terminal is not fully functional
-  (press RETURN) 
commit b737624db5df34cdbb882f459ee96667e817e793
Author: Jason E. Aten <j.e.aten@gmail.com>
Date:   Sat Jun 25 20:38:10 2016 -0500

    add --http-addr flag
    
    For security, the --http-addr flags makes the Admin UI
    bind an IP address that can be private or distinct from
    the cluster. If --host is given, we default to that
    address if --http-addr is unspecified.
    
    Fixes #7474.

commit 5695764788c5000082af998fda9ec8211d15138b
Merge: 35126dc a12e9d5
Author: kena <knz@users.noreply.github.com>
Date:   Fri Jun 24 21:13:17 2016 +0200

    Merge pull request #7447 from knz/envutil-threadsafe
    
    envutil: protect concurrent accesses to the env var cache.

commit 35126dc63f62f06b0df9a05ac3b8a2636b0991a9
Merge: 549d9b5 9e99e28
:q
jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ g++ --version
g++ (Ubuntu 4.9.3-8ubuntu2~14.04) 4.9.3
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ gcc --version
gcc (Ubuntu 4.9.3-8ubuntu2~14.04) 4.9.3
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

jaten@crdev:~/go/src/github.com/cockroachdb/cockroach$ 
~~~

--- a/build/builder.sh
+++ b/build/builder.sh
@@ -79,8 +79,9 @@ cockroach_toplevel="$(dirname $(cd $(dirname $0); pwd))"
 docker run -i ${tty-} ${rm} \
   --volume="${gopath0}/src:/go/src" \
   --volume="${gopath0}/pkg:/go/pkg" \
-  --volume="${gopath0}/pkg/linux_amd64_race:/usr/src/go/pkg/linux_amd64_race" \
-  --volume="${gopath0}/bin/linux_amd64:/go/bin" \
+  --volume="${gopath0}/pkg/docker_amd64:/usr/src/go/pkg/linux_amd64" \
+  --volume="${gopath0}/pkg/docker_amd64_race:/usr/src/go/pkg/linux_amd64_race" \
+  --volume="${gopath0}/bin/docker_amd64:/go/bin" \
   --volume="${HOME}/.jspm:/root/.jspm" \
   --volume="${HOME}/.npm:/root/.npm" \
   --volume="${cockroach_toplevel}:/go/src/github.com/cockroachdb/cockroach" \

--- a/build/circle-deps.sh
+++ b/build/circle-deps.sh
@@ -154,7 +154,7 @@ if is_shard 2; then
     done
 
     cmds=$(grep '^cmd ' GLOCKFILE | grep -v glock | awk '{print $2}' | awk -F/ '{print $NF}')
-    path="${gopath0}/bin/linux_amd64"
+    path="${gopath0}/bin/docker_amd64"
     time ssh node0 mkdir -p "$path"
     time ssh node0 sudo chown "${USER}.${USER}" "$path"
     (time cd "$path" && rsync ${cmds} node0:"${path}")
@@ -165,7 +165,7 @@ if is_shard 1; then
   # We might already be on shard 0 if we're running without
   # parallelism. Avoid deleting our build output in that case.
   if ! is_shard 0; then
-    dir="${gopath0}/pkg/linux_amd64_race"
+    dir="${gopath0}/pkg/docker_amd64_race"
     time ssh node0 sudo chown -R "${USER}.${USER}" "${dir}"
     time rsync -a --delete "${dir}/" node0:"${dir}"
   fi
@@ -191,7 +191,7 @@ if is_shard 0; then
   # building an overlapping set of packages.
   if ! is_shard 2; then
     node=$((2 % $CIRCLE_NODE_TOTAL))
-    dir="${gopath0}/pkg/linux_amd64"
+    dir="${gopath0}/pkg/docker_amd64"
     time sudo chown -R "${USER}.${USER}" "${dir}"
     time rsync -au node${node}:"${dir}/" "${dir}"
   fi

